service/controller/resource/controlplanestatus/resource.go

```diff
@@ -70,7 +71,7 @@ func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
 		}
 	}
 	masterNodes, err := r.nodeCount.MasterCount(ctx, cr)
-	if nodecount.IsTenantClusterInitialized(err) {
+	if tenantclient.IsNotAvailable(err) || nodecount.IsTenantClusterNotInitialized(err) {
```

Comment:
`nodecount.IsTenantClusterNotInitialized` is still fishy. We should not need it. When the client is not there `tenantclient` returns an error which we handle already. So I still think the `nodecount` error is redundant. I am not sure if I communicate my thought process well.